client/service/src/chain_ops.rs

```diff
@@ -298,4 +299,45 @@ impl<
 			Err(e) => Box::pin(future::err(format!("Error reading block: {:?}", e).into())),
 		}
 	}
+
+	fn export_raw_state(
+		&self,
+		block: Option<BlockId<Self::Block>>,
+	) -> Result<Storage, Error> {
+		let block = block.unwrap_or_else(
+			|| BlockId::Hash(self.client.usage_info().chain.best_hash)
+		);
+
+		let key = StorageKey(Vec::new());
+		let mut top_storage = self.client.storage_pairs(&block, &key)?;
+		let mut children_default = HashMap::new();
+
+		// Remove all default child storage roots from the top storage and collect the child storage
+		// pairs.
+		while let Some(pos) = top_storage
```

Comment:
@cheme it would be nice if you could take a look here, if I have done this child value collection correctly.